### ISSUE TITLE
Fatal error in install.php after installing contao 3.5.1 via git

### ISSUE THREAD

**Issue by bezin** [contao/core#7937]
Hei,

I just wanted to install the new 3.5.1 release via git. I cloned the repo, I ran composer install and wanted to run the install tool. I just get an Fatal Error
```
Fatal error: Class 'Database\' not found in C:\development\wamp\www\contao351\system\modules\core\library\Contao\Database.php on line 165
```
Win 8.1 / Wamp with PHP 5.5.12.

**Comment by leofeyer** [comment]
What database driver did you select in the install tool?

**Comment by bezin** [comment]
I don't get there. I get the fatal error the very first time I go to /contao/install.php

**Comment by leofeyer** [comment]
Check the local configuration file then.

**Comment by bezin** [comment]
No local config yet at that point. It works if I manually create a localconfig.php and set $GLOBALS['TL_CONFIG']['dbDriver'] = 'Mysqli'; though.

**Comment by fritzmg** [comment]
Hm, I was able to reproduce it on my local system.

* Windows 7 x64
* XAMPP 1.8.2
* PHP 5.4.19

All I did was
```
git clone https://github.com/contao/core.git
cd core
php composer.phar install --prefer-dist
```
as per the current [README](https://github.com/contao/core/blob/3.5.1/README.md). I haven't yet debugged why the class cannot be found.

**Comment by bezin** [comment]
The class cannot be found because $GLOBALS['TL_CONFIG']['dbDriver'] is null. Therefore, 
this line https://github.com/contao/core/blob/3.5.1/system/modules/core/library/Contao/Database.php#L164 sets $strClass to 'Database\' and line 165 causes the fatal error.

**Comment by fritzmg** [comment]
The same happens when using a distribution archive like this one: [contao-3.5.1.zip](https://github.com/contao/core/releases/download/3.5.1/contao-3.5.1.zip)

**Comment by fritzmg** [comment]
A clean installation still works in Contao 3.5.0. The problem (difference) in Contao 3.5.1 is this line: https://github.com/contao/core/blob/3.5.1/system/modules/core/controllers/BackendInstall.php#L33
```php
	public function __construct()
	{
		parent::__construct();

		…
	}
```
And in the constructor of the parent, i.e. `\Backend` this happens: https://github.com/contao/core/blob/3.5.1/system/modules/core/classes/Backend.php#L30
```php
		$this->import('Database');
```
Which causes the Install Tool to fail without a proper Database configuration (which there will be none on a fresh install).

**Comment by bezin** [comment]
Just checked the 3.5.0 installer as well. I can confirm the explanation of @fritzmg

**Comment by fritzmg** [comment]
imho this problem should be solved more generally. For example, if your `localconfig.php` gets somehow "lost" in a running installation, you might not be able to run the Install Tool either, because some extensions may have some `initializeSystem` hooks which also might call `\Database::getInstance()` at some point (directly or indirectly).

Maybe it would be better to simply return a "dummy" database object, whenever the config is incomplete.

**Comment by bezin** [comment]
@fritzmg: That actually happend many times with the dma_elementgenerator. I still haven't figure out in depth what the problem was, but I got that fatal error several times with that extension because the localconfig.php "got somehow lost" and I wasn't able to call the installer again.

**Comment by leofeyer** [comment]
Fixed in 636955015aba557ef3b0001486f991fae9cc70e5. Will release version 3.5.2 in a few minutes.

**Comment by bezin** [comment]
Thanks a lot!
